keras/engine/topology.py

```diff
@@ -284,10 +284,14 @@ def __init__(self, **kwargs):
 
         # these properties will be set upon call of self.build(),
         # which itself will be called upon self.add_inbound_node if necessary.
-        self.trainable_weights = []
-        self.non_trainable_weights = []
-        self.regularizers = []
-        self.constraints = {}  # dict {tensor: constraint instance}
+        if not hasattr(self, 'trainable_weights'):
+                self.trainable_weights = []
+        if not hasattr(self, 'non_trainable_weights'):
+                self.non_trainable_weights = []
+        if not hasattr(self, 'regularizers'):
+                self.regularizers = []
+        if not hasattr(self, 'constraints'):
+                self.constraints = {}  # dict {tensor: constraint instance}
```

Comment:
Better to leave this part of the code unchanged and implement a property setter on the wrapper.